src/libs/TransactionUtils.js

```diff
@@ -79,12 +80,18 @@ function getUpdatedTransaction(transaction, transactionChanges, isFromExpenseRep
     }
     if (_.has(transactionChanges, 'created')) {
         updatedTransaction.modifiedCreated = transactionChanges.created;
+        shouldStopSmartscan = true;
     }
     if (_.has(transactionChanges, 'amount')) {
         updatedTransaction.modifiedAmount = isFromExpenseReport ? -transactionChanges.amount : transactionChanges.amount;
+        shouldStopSmartscan = true;
     }
     if (_.has(transactionChanges, 'currency')) {
         updatedTransaction.modifiedCurrency = transactionChanges.currency;
+        shouldStopSmartscan = true;
+    }
+    if (shouldStopSmartscan && _.has(transaction, 'receipt') && !_.isEmpty(transaction.receipt) && lodashGet(transaction, 'receipt.state') !== CONST.IOU.RECEIPT_STATE.OPEN) {
+        updatedTransaction.receipt.state = CONST.IOU.RECEIPT_STATE.OPEN;
```

Comment:
If the transaction state is set to `CONST.IOU.RECEIPT_STATE.OPEN` will that automatically stop a receipt from SmartScanning? If not, how are we going about canceling a smart scan if one has already started, or if we will start one as soon as the user gets back online?

Reply:
I think these are not considerations for this PR specifically. This is only when the transaction is being edited and when the edit API request is processed we will stop the smartscan. 

Otherwise we will have to handle the overwrited changes with Red brick road but that can be in next prs

Reply:
Sorry, @mountiny I don't understand.

We are creating a variable here called `shouldStopSmartScan`. That leads me to believe that if `shouldStopSmartScan` is `false`, the receipt will be smart scanned. If it is set to `true`, then the receipt will not be smart scanned.

The only thing I see in this PR that is being updated that might possible stop the smart scanning is the changing of the `receipt.state` to `OPEN`.

Is that what would cause the smart scan to stop?

Reply:
Gotcha, the smartscan will be stopped in the EditMoneyRequest command in Auth, whenver the [change is detected ](https://github.com/Expensify/Auth/blob/8e4d9fac78410ee42b54ec9244b8a549351ac5d6/auth/command/EditMoneyRequest.cpp#L168)

You can see it [here](https://github.com/Expensify/Auth/blob/8e4d9fac78410ee42b54ec9244b8a549351ac5d6/auth/command/EditMoneyRequest.cpp#L216-L220). 

What is doen in this PR is just optimistically reflecting the same change we do in the backend in the Offline first fashion